src/auxil.c

```diff
@@ -1046,4 +1060,19 @@ c_int validate_settings(const OSQPSettings *settings) {
   return 0;
 }
 
+c_int memory_alloc_error(void) {
```

Comment:
Two suggestions:

1. I would combine these functions in something for general error handling to make it more consistent. An option is to create an `error.c` file and put there an unifying error function getting all error codes and have its declaration in `glop_opts.h` with error codes in `constants.h`. The function can be something like `c_int osqp_error(c_int ERROR_CODE);` with all the cases. It is more readable and maintainable IMHO.

2. We can create an enum with all error codes and a mapping between the enum errors and string constants like we did for linear system solvers:
```c
static const char *LINSYS_SOLVER_NAME[] = {
  "qdldl", "mkl pardiso"
};
```
That code could go into the `error.c` file after defining it with `extern` in `constants.h` as I did in the last commit for `LINSYS_SOLVERS`. I would avoid defining global variables in headers.

What do you think @gbanjac @goulart-paul?